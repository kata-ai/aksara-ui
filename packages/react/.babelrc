{
  "presets": [
    [
      "@babel/preset-env",
      {
        "targets": {
          "browsers": ["last 2 chrome versions", "last 2 firefox versions"]
        }
      }
    ],
    "@babel/preset-react",
    "@babel/preset-typescript"
  ],
  "plugins": [
    "@babel/plugin-proposal-class-properties",
    "@babel/plugin-proposal-optional-chaining",
    "@babel/plugin-proposal-nullish-coalescing-operator",
    "babel-plugin-styled-components"
  ],
  "env": {
    "test": {
      "plugins": ["@babel/plugin-transform-modules-commonjs"]
    }
  }
}
