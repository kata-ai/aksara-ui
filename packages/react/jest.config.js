module.exports = {
  collectCoverageFrom: [
    'src/**/*.{ts,tsx}',
    '!src/**/*.stories.{ts,tsx}',
    '!src/**/*.story.{ts,tsx}',
    '!src/**/index.{ts,tsx}',
    '!src/stories/**/*.{ts,tsx}',
    '!src/utils/storybook/**/*.{ts,tsx}',
    '!src/index.{ts,tsx}',
  ],
  coverageReporters: ['json', 'lcov'],
  moduleDirectories: ['node_modules', '<rootDir>/src'],
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json'],
  modulePathIgnorePatterns: ['lib', 'lib-es6'],
  setupFilesAfterEnv: ['<rootDir>/test/setupTests.ts'],
  testPathIgnorePatterns: ['<rootDir>/dist/'],
  transform: {
    '^.+\\.(js|jsx|ts|tsx|mjs)$': 'babel-jest',
    '^.+\\.css$': '<rootDir>/config/jest/cssTransform.js',
    '^(?!.*\\.(js|jsx|ts|tsx|mjs|css|json)$)': '<rootDir>/config/jest/fileTransform.js',
  },
};
