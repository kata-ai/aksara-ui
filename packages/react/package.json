{
  "name": "@aksara-ui/react",
  "version": "1.0.1-canary.4",
  "description": "React implementation of Aksara Design System, from Kata.ai.",
  "main": "dist/index.js",
  "module": "dist/core.esm.js",
  "typings": "dist/index.d.ts",
  "aksara:src": "./src/index.ts",
  "repository": "https://github.com/kata-ai/wicara.git",
  "author": "Kata.ai <dev@kata.ai>",
  "license": "Apache-2.0",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "tsdx build --transpileOnly --tsconfig tsconfig.build.json",
    "build-storybook": "build-storybook --quiet",
    "clean": "rimraf dist",
    "format": "prettier --write \"src/**/*.{js,jsx,ts,tsx}\"",
    "lint": "eslint \"src/**/*.{js,jsx,ts,tsx}\"",
    "prepublish": "yarn build",
    "report-coverage": "codecov -f ./coverage/*.json",
    "start": "tsdx watch --tsconfig tsconfig.build.json --verbose",
    "storybook": "start-storybook -p 9001",
    "storybook:build": "build-storybook --quiet",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "type-check": "tsc --emitDeclarationOnly"
  },
  "dependencies": {
    "@aksara-ui/core": "^1.0.1-canary.1",
    "@popperjs/core": "^2.10.2",
    "@reach/accordion": "^0.16.1",
    "@reach/visually-hidden": "^0.16.0",
    "@styled-system/css": "^5.1.5",
    "@styled-system/should-forward-prop": "^5.1.5",
    "@styled-system/theme-get": "^5.1.2",
    "@types/styled-system": "^5.1.13",
    "@types/styled-system__css": "^5.0.16",
    "@types/styled-system__theme-get": "^5.0.1",
    "clsx": "^1.1.1",
    "csstype": "^3.0.9",
    "deepmerge": "^4.2.2",
    "focus-trap": "^6.6.0",
    "polished": "^4.1.3",
    "react-fast-compare": "^3.2.0",
    "react-popper": "^2.2.5",
    "react-popper-tooltip": "^4.3.0",
    "react-transition-group": "^4.4.2",
    "styled-system": "^5.1.4",
    "tslib": "^2.3.1"
  },
  "peerDependencies": {
    "@aksara-ui/icons": "^1.0.0-canary.0",
    "react": "^16.10.0 || ^17.0.0",
    "react-dom": "^16.10.0 || ^17.0.0",
    "styled-components": "^5.1.0"
  },
  "devDependencies": {
    "@aksara-ui/icons": "^1.0.0",
    "@babel/core": "^7.15.0",
    "@storybook/addon-actions": "^6.3.7",
    "@storybook/addon-essentials": "^6.3.7",
    "@storybook/addon-knobs": "^6.3.0",
    "@storybook/addon-links": "^6.3.7",
    "@storybook/react": "^6.3.7",
    "@storybook/theming": "^6.3.7",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^12.0.0",
    "@types/jest": "^26.0.24",
    "@types/react-transition-group": "^4.4.2",
    "@types/styled-components": "^5.1.12",
    "babel-loader": "^8.2.2",
    "babel-plugin-styled-components": "^1.13.2",
    "babel-preset-react-app": "^10.0.0",
    "codecov": "^3.8.3",
    "copee": "^1.0.6",
    "jest": "^26.6.3",
    "jest-styled-components": "^7.0.5",
    "react": "^17.0.2",
    "react-docgen-typescript-loader": "^3.6.0",
    "react-dom": "^17.0.2",
    "styled-components": "^5.3.0"
  }
}
