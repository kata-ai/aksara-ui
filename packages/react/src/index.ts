export * from './aksara-provider';
export * from './components';
export * from './layout';
export * from './theme';
export * from './typography';
export * from '@aksara-ui/helpers';
