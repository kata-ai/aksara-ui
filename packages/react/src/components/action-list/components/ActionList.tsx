import * as React from 'react';
import { Stack, StackProps } from '../../../layout';

export interface ActionListProps extends StackProps, React.ComponentPropsWithoutRef<'div'> {}

const ActionList = React.forwardRef<HTMLDivElement, ActionListProps>(({ children, ...rest }, ref) => {
  return (
    <Stack
      display="block"
      spacing="xxs"
      borderRadius="lg"
      paddingY="sm"
      backgroundColor="greylight01"
      ref={ref}
      _focusVisible={{
        outline: '0px',
      }}
      {...rest}
    >
      {children}
    </Stack>
  );
});

export default ActionList;
