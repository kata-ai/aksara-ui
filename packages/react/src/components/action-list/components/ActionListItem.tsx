import * as React from 'react';
import { Box, BoxProps } from '../../../layout';
import { CSSObject, useComponentStyles } from '../../../system';
import { theme } from '../../../theme';
import { Text } from '../../../typography';

export type ActionListItemVariant = 'default' | 'destructive';
export interface ActionListItemProps extends Omit<BoxProps, 'sx'>, React.ComponentPropsWithoutRef<'div'> {
  className?: string;
  style?: React.CSSProperties;
  containerStyle?: CSSObject;
  contentStyle?: CSSObject;
  isActive?: boolean;
  children?: React.ReactNode;
  disabled?: boolean;
  variant?: ActionListItemVariant;
}

const ActionListItem = React.forwardRef<HTMLDivElement, ActionListItemProps>(
  ({ style, children, isActive, disabled, containerStyle, variant = 'default', contentStyle, ...rest }, ref) => {
    const styles = useComponentStyles('actionListItem', { isActive, variant });
    const renderLabel = () => {
      if (typeof children === 'string' || typeof children === 'number') {
        return <Text scale={300}>{children}</Text>;
      }
      return children;
    };
    return (
      <Box
        aria-disabled={disabled}
        data-disabled={disabled}
        ref={ref}
        sx={{ ...styles, ...containerStyle }}
        style={style}
        {...rest}
      >
        {isActive && (
          // Indicator
          <Box width={4} position="absolute" left="0" top="0" height="100%" backgroundColor={theme.colors.blue07} />
        )}
        <Box py="xs" px="sm" sx={{ ...contentStyle }}>
          {renderLabel()}
        </Box>
      </Box>
    );
  }
);

export default ActionListItem;
