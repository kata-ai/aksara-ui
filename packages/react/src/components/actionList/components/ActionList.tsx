import { StackProps } from 'packages/react/dist';
import * as React from 'react';
import { Stack } from '../../../layout';

export interface ActionListProps extends StackProps, React.ComponentPropsWithoutRef<'div'> {}

const ActionList = React.forwardRef<HTMLDivElement, ActionListProps>(({ children, ...rest }, ref) => {
  return (
    <Stack spacing="xxs" borderRadius="lg" paddingY="sm" backgroundColor="greylight01" ref={ref} {...rest}>
      {children}
    </Stack>
  );
});

export default ActionList;
