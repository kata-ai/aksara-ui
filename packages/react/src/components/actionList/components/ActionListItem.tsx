import * as React from 'react';
import { Box, BoxProps } from '../../../layout';
import { useComponentStyles } from '../../../system';
import { theme } from '../../../theme';
import { Text } from '../../../typography';

export interface ActionListItemProps extends BoxProps, React.ComponentPropsWithoutRef<'div'> {
  className?: string;
  style?: React.CSSProperties;
  isActive?: boolean;
  children?: React.ReactNode;
  disabled?: boolean;
}

const ActionListItem = React.forwardRef<HTMLDivElement, ActionListItemProps>(
  ({ style, children, isActive, disabled, ...rest }, ref) => {
    const styles = useComponentStyles('actionListItem', { isActive });
    const renderLabel = () => {
      if (typeof children === 'string' || typeof children === 'number') {
        return (
          <Text scale={300} marginLeft={isActive ? '-4px' : ''}>
            {children}
          </Text>
        );
      }
      return children;
    };
    return (
      <Box aria-disabled={disabled} data-disabled={disabled} ref={ref} sx={styles} {...rest}>
        {isActive && (
          // Indicator
          <Box width={4} position="absolute" left="0" top="0" height="100%" backgroundColor={theme.colors.blue07} />
        )}
        <Box py="xs" px="sm" style={style}>
          {renderLabel()}
        </Box>
      </Box>
    );
  }
);

export default ActionListItem;
