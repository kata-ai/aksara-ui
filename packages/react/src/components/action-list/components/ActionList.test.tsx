import * as React from 'react';
import { render } from '@testing-library/react';

import ActionList from './ActionList';
import ActionListItem from './ActionListItem';

describe('components/ActionList', () => {
  describe('<ActionList />', () => {
    test('renders correctly', () => {
      const { container } = render(<ActionList>example</ActionList>);

      expect(container.firstChild).toBeInTheDocument();
      expect(container.firstChild).toMatchSnapshot();
    });

    test('renders actionList with item correctly', () => {
      const { container } = render(
        <ActionList>
          <ActionListItem isActive>Label 1</ActionListItem>
          <ActionListItem disabled>Label 2</ActionListItem>
          <ActionListItem disabled>Label 2</ActionListItem>
        </ActionList>
      );

      expect(container.firstChild).toBeInTheDocument();
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
