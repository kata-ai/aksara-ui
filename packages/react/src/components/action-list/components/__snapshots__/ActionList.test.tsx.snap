// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/ActionList <ActionList /> renders actionList with item correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: block;
  border-radius: lg;
  padding-top: sm;
  padding-bottom: sm;
  background-color: greylight01;
}

.c0 > :not([hidden]) ~ :not([hidden]) {
  margin-top: xxs;
}

.c0:focus-visible {
  outline: 0px;
}

.c1 {
  width: 4px;
  position: absolute;
  left: 0;
  top: 0;
  height: 100%;
  background-color: #006FE6;
}

.c2 {
  padding-top: xs;
  padding-bottom: xs;
  padding-left: sm;
  padding-right: sm;
}

<div
  class="c0"
>
  <div
    class=""
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    >
      <span
        class="Text-sc-18vu504-0"
      >
        Label 1
      </span>
    </div>
  </div>
  <div
    aria-disabled="true"
    class=""
    data-disabled="true"
  >
    <div
      class="c2"
    >
      <span
        class="Text-sc-18vu504-0"
      >
        Label 2
      </span>
    </div>
  </div>
  <div
    aria-disabled="true"
    class=""
    data-disabled="true"
  >
    <div
      class="c2"
    >
      <span
        class="Text-sc-18vu504-0"
      >
        Label 2
      </span>
    </div>
  </div>
</div>
`;

exports[`components/ActionList <ActionList /> renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: block;
  border-radius: lg;
  padding-top: sm;
  padding-bottom: sm;
  background-color: greylight01;
}

.c0 > :not([hidden]) ~ :not([hidden]) {
  margin-top: xxs;
}

.c0:focus-visible {
  outline: 0px;
}

<div
  class="c0"
/>
`;
